sql/pgwire/v3.go

```diff
@@ -337,12 +337,12 @@ func (c *v3Conn) handleParse(buf *readBuffer) error {
 	for k, v := range args {
 		i, err := strconv.Atoi(k)
 		if err != nil {
-			return c.sendError(fmt.Sprintf("non-integer parameter name: %s", k))
+			return c.sendError(fmt.Sprintf("non-integer parameter: %s", k))
 		}
 		// ValArgs are 1-indexed, pq.inTypes are 0-indexed.
 		i--
-		if len(pq.inTypes) <= i {
-			return c.sendError("internal error: leftover valargs")
+		if i < 0 || len(pq.inTypes) <= i {
+			return c.sendError(fmt.Sprintf("there is no parameter $%s", k))
```

Comment:
Hmm, ok. I changed the way this loop works so that we grow the `pq.inTypes` array as necessary instead of assuming it is the same size as `len(args)`. And I've changed the error for unable to infer parameter type to match postgres.